### PR TITLE
backintime: fix backintime-qt_polkit

### PR DESCRIPTION
**Pull request by stephen-huan**
## Description of changes

Fix running `backintime-qt_polkit`, which is necessary if one wants to back up their entire system, e.g.
(and should be slightly more secure than directly running `sudo backintime-qt`).

Most changes are straightforward substitutions of hardcoded paths. 

Note that `polkit` is necessary since `backintime-qt_polkit` is just a thin wrapper which uses `pkexec`, i.e.
```shell
#!/nix/store/lf0wpjrj8yx4gsmw2s3xfl58ixmqk8qa-bash-5.2-p15/bin/sh

if [ "x$XDG_SESSION_TYPE" = "xwayland" ]; then
    PREFIX="env QT_QPA_PLATFORM=wayland-egl XDG_RUNTIME_DIR=$XDG_RUNTIME_DIR"
else
    PREFIX=""
fi

pkexec --disable-internal-agent $PREFIX "/usr/bin/backintime-qt" "$@"
```

Also note that the application sometimes segfaults on close, this is a known [upstream issue](https://github.com/bit-team/backintime/issues/1095).

@bjornfor 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/sync/backintime/qt.nix [hunk 1]
```diff
@@ -22,5 +22,27 @@ mkDerivation {
...
+      --replace "/usr/bin/nice" "${coreutils}/bin/nice" \
+      --replace "/usr/bin/ionice" "${util-linux}/bin/ionice"
+
+    substituteInPlace "$out/share/dbus-1/system-services/net.launchpad.backintime.serviceHelper.service" \
+      --replace "/usr/bin/python3" "${python'}/bin/python3" \
+      --replace "/usr/share/backintime" "$out/share/backintime"
+
+    substituteInPlace "$out/bin/backintime-qt_polkit" \
+      --replace "/usr/bin/backintime-qt" "$out/bin/backintime-qt"
+    wrapProgram "$out/bin/backintime-qt_polkit" \
```
Comment (bjornfor) [inline pkgs/applications/networking/sync/backintime/qt.nix:45]: I think a blank line above wrapProgram would be good here (and in line with the rest of the changes).

File: pkgs/applications/networking/sync/backintime/qt.nix [hunk 2]
```diff
@@ -22,5 +22,27 @@ mkDerivation {
...
       --prefix PATH : "${backintime-common}/bin:$PATH"
+
+    substituteInPlace "$out/share/polkit-1/actions/net.launchpad.backintime.policy" \
+      --replace "/usr/bin/backintime-qt" "$out/bin/backintime-qt"
+
+    substituteInPlace "$out/share/applications/backintime-qt-root.desktop" \
+      --replace "/usr/bin/backintime-qt" "backintime-qt"
+
+    substituteInPlace "$out/share/backintime/qt/serviceHelper.py" \
+      --replace "'which'" "'${which}/bin/which'" \
```
Comment (bjornfor) [inline pkgs/applications/networking/sync/backintime/qt.nix:33]: Maybe replace occurrences like `${drv}/bin/name` with `${lib.getBin drv}/bin/name`?

Comment (stephen-huan) [reply pkgs/applications/networking/sync/backintime/qt.nix]: fixed

File: pkgs/applications/networking/sync/backintime/qt.nix (hunk 1 above)
Comment (stephen-huan) [reply pkgs/applications/networking/sync/backintime/qt.nix:46]: fixed

**Review by bjornfor — APPROVED** [review]

**Comment by bjornfor** [comment]
Thanks!

**Pull request merged**